Otec-iOS/Otec/CreateNewspaperViewController.swift

```diff
@@ -9,27 +9,59 @@
 import UIKit
 
 class CreateNewspaperViewController: UIViewController {
+    @IBOutlet weak var nameTextfield: UITextField!
+    @IBOutlet weak var descriptionTextView: UITextView!
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        // Do any additional setup after loading the view.
+        let createButton = UIBarButtonItem(title: "Create", style: .Plain, target: self, action:#selector(createNewspaper))
+        self.navigationItem.rightBarButtonItem = createButton
     }
-
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    
+    private func pushFeedsViewController(animated animated: Bool) {
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        let viewController = storyboard.instantiateViewControllerWithIdentifier("NewsFeedViewController") as! NewsFeedViewController
+        self.navigationController?.pushViewController(viewController, animated: animated)
     }
     
-
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
+    func createNewspaper(sender: AnyObject) {
+        if !self.checkFieldsEmptiness(self.nameTextfield, descriptionTextView: self.descriptionTextView) {
+            print ("name or description cannot be empty")
+            return
+        }
+        
+        let newspaperDictionary = ["name":self.nameTextfield.text!,
+                                   "description":self.descriptionTextView.text!]
+        
+        guard let newspaper = try? Newspaper.init(dictionary:newspaperDictionary) else {
+            print("error")
+            return
+        }
+        let future = NewspaperRepository().create(newspaper)
+        future.start() { result in
+            switch result {
+            case .Success(let newspaper):
+                dispatch_async(dispatch_get_main_queue()) {
+                    NSUserDefaults.standardUserDefaults().setObject(newspaper.id, forKey:Constants.newspaperUserDefaultsKey)
+                    self.pushFeedsViewController(animated: true)
+                }
+            case .Failure(let error):
+                print ("error : \(error)")
+            }
+        }
     }
-    */
-
+    
+    private func checkFieldsEmptiness(nameTextfield: UITextField, descriptionTextView: UITextView) -> Bool{
```

Comment:
Let's discuss on how to improve this function with a protocol-oriented approach ;)